src/modm/platform/dma/stm32/dma.hpp.in

```diff
@@ -34,107 +25,366 @@ namespace platform
+		 */
+		static void
+		stop()
+		{
+			ChannelHal::stop();
+		}
+
+		/**
+		 * Get the direction of the data transfer
+		 */
+		static DataTransferDirection
+		getDataTransferDirection()
+		{
+			return ChannelHal::getDataTransferDirection();
+		}
+
+		/**
+		 * Set the source and destination address of the DMA channel
+		 *
+		 * Requires to set the direction of the DMA channel first
+		 *
+		 * @param[in] sourceAddress Source address of DMA transfer
+		 * @param[in] destinationAddress Destination address of DMA transfer
+		 */
+		static void
+		setAddresses(uint32_t sourceAddress, uint32_t destinationAddress)
+		{
+			ChannelHal::setAddresses(sourceAddress, destinationAddress);
+		}
+		/**
+		 * Set the source address of the DMA channel
+		 *
+		 * @param[in] address Source address
+		 */
+		static void
+		setSourceAddress(uint32_t address)
+		{
+			ChannelHal::setSourceAddress(address);
+		}
+		/**
+		 * Set destination address of DMA channel
+		 *
+		 * @param[in] address Destination address
+		 */
+		static void
+		setDestinationAddress(uint32_t address)
+		{
+			ChannelHal::setDestinationAddress(address);
+		}
+
+		/**
+		 * Enable/disable memory increment
+		 *
+		 * When enabled, the memory address is incremented by the size of the data
+		 * (e.g. 1 for byte transfers, 4 for word transfers) after the transfer
+		 * completed.
+		 *
+		 * @param[in] increment Enable/disable
+		 */
+		static void
+		setMemoryIncrementMode(bool increment)
+		{
+			ChannelHal::setMemoryIncrementMode(increment);
+		}
+		/**
+		 * Enable/disable peripheral increment
+		 *
+		 * When enabled, the peripheral address is incremented by the size of the data
+		 * (e.g. 1 for byte transfers, 4 for word transfers) after the transfer
+		 * completed.
+		 *
+		 * @param[in] increment Enable/disable
+		 */
+		static void
+		setPeripheralIncrementMode(bool increment)
+		{
+			ChannelHal::setPeripheralIncrementMode(increment);
+		}
+
+		/**
+		 * Set the length of data to be transfered
+		 */
+		static void
+		setDataLength(std::size_t length)
+		{
+			ChannelHal::setDataLength(length);
+		}
+
+		/**
+		 * Set the IRQ handler for transfer errors
+		 *
+		 * The handler will be called from the channels IRQ handler function
+		 * when the IRQ status indicates an error occured.
+		 */
+		static void
+		setTransferErrorIrqHandler(IrqHandler irqHandler)
+		{
+			transferError = irqHandler;
+		}
+		/**
+		 * Set the IRQ handler for transfer complete
+		 *
+		 * Called by the channels IRQ handler when the transfer is complete.
+		 */
+		static void
+		setTransferCompleteIrqHandler(IrqHandler irqHandler)
+		{
+			transferComplete = irqHandler;
+		}
+		/**
+		 * Set the peripheral that operates the channel
+		 */
+		template <DmaBase::Request dmaRequest>
+		static void
+		setPeripheralRequest()
+		{
+			DMA_Request_TypeDef *DMA_REQ = reinterpret_cast<DMA_Request_TypeDef *>(ControlHal::DMA_CSEL);
+			DMA_REQ->CSELR &= ~(0x07 << (uint32_t(ChannelID) * 4));
```

Comment:
Is there a reason why only the lower 3 bits of the 4 bit value are cleared here?

Reply:
The data sheet list only value 0 to 7 as valid values, so that's the reason. Fixed.